File: 2sem/javaprogramy/Lista2/Trójkąt Pascala java/TrojkatPascala.java
import java.util.ArrayList;
public class TrojkatPascala {
    
    int n;
    ArrayList<Integer> tab = new ArrayList<>();

    TrojkatPascala (int n){
        utworzTablice(n);
        }

    void utworzTablice(int n){
        for (int k=0;k<=n;k++){
            int wynik= silnia(n)/(silnia(k)*silnia(n-k));
            tab.add(wynik);
        }
    }

    int element (int m){
        if (m>=0&&m<tab.size()){
            return tab.get(m);
        } else return -1;
    }

    int silnia (int x){
        int wynik=1;
        for (int i=1;i<=x;i++){
            wynik*=i;
        }
        return wynik;
    }
}

File: 2sem/javaprogramy/Lista4/Zadanie3/TrojkatPascala.cpp
#include <iostream>
#include <vector>
#include "funkcje.hpp"

using namespace std;

TrojkatPascala::TrojkatPascala(int wiersz){
    utworzTab(wiersz);
}

TrojkatPascala::~TrojkatPascala(){

}

int TrojkatPascala::silnia (int n){
    int wynik=1;
    for (int i=1;i<=n;i++){
        wynik*=i;
    }
    return wynik;
}

void TrojkatPascala::utworzTab(int n){
    int wynik;
    for (int i=0;i<=n;i++){
        wynik=silnia(n)/(silnia(i)*silnia(n-i));
        tab.push_back(wynik);
    }
}

int TrojkatPascala::element(int ktory){
    if (ktory>=0&& ktory<tab.size()) return tab[ktory];
    else return -1;
}

